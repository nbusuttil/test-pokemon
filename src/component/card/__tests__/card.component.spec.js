import React from 'react';
import { shallow } from 'enzyme';

import Card from '../card.component';

describe('Card component', () => {
  let props;

  beforeEach(() => {
    props = {
      pokemon: {
        id: 2,
        price: 7,
        quantity: 0,
        name: 'spearow',
        url: 'https://pokeapi.co/api/v2/pokemon/21/'
      },
      action: jest.fn(),
      openModal: jest.fn(),
      titleAction: 'Add to cart',
      addQuantity: jest.fn(),
      removeQuantity: jest.fn(),
      titleAdd: '+',
      titleRemove: '-',
      modifier: false,
      isShop: false
    };
  });
  const getWrapper = () => shallow(<Card {...props} />);

  it('render without crashing', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
