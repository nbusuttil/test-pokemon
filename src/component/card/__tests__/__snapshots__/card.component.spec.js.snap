// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render without crashing 1`] = `
<div
  className="card"
>
  <button
    className="card__picture"
    onClick={[Function]}
  >
    <img
      alt="spearow"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
    />
  </button>
  <h1
    className="card__title"
  >
    <span>
      #
      2
    </span>
     
    spearow
  </h1>
  <div
    className="card__actions"
  >
    <div
      className="card__actions-item"
    >
      <button
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
  </div>
  <div
    className="card__info"
  >
    <p
      className="card__info-price"
    >
      <span>
        Prix:
      </span>
       
      7
       €
    </p>
  </div>
</div>
`;
